src/ray/pubsub/publisher.cc

```diff
@@ -22,22 +22,24 @@ namespace pubsub {
 
 namespace pub_internal {
 
-bool BasicEntityState::Publish(std::shared_ptr<rpc::PubMessage> msg) {
+bool EntityState::Publish(std::shared_ptr<rpc::PubMessage> msg) {
   if (subscribers_.empty()) {
     return false;
   }
-  for (auto &[id, subscriber] : subscribers_) {
-    subscriber->QueueMessage(msg);
-  }
-  return true;
-}
 
-bool CappedEntityState::Publish(std::shared_ptr<rpc::PubMessage> msg) {
-  if (subscribers_.empty()) {
+  const int64_t message_size = msg->ByteSizeLong();
+
+  if (message_size > max_message_size_bytes_) {
+    RAY_LOG_EVERY_N_OR_DEBUG(WARNING, 10000)
```

Comment:
I'd suggest to keep this a debug (there's nothing really to warn about)

Reply:
I'd prefer to keep it as warning because we are dropping a message. That can have unknown side effects if the message dropped was critical. By the way, this only warns every 10,000th time, so I think it will not be too verbose.